src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -165,7 +165,26 @@ private void ValidateState()
                 case FrameStreamState.Closed:
                     throw new ObjectDisposedException(nameof(FrameRequestStream));
                 case FrameStreamState.Aborted:
-                    throw new IOException("The request has been aborted.");
+                    throw new OperationCanceledException("The request has been aborted.");
+            }
+        }
+
+        private void ValidateStateAsync(CancellationToken cancellationToken)
+        {
+            if (_state == FrameStreamState.Closed)
+            {
+                throw new ObjectDisposedException(nameof(FrameRequestStream));
+            }
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
No, but will throw `OperationCanceledException` cancelling a task or tcs with throw `TaskCanceledException`; I assume to differentiate whether it's throwing on sync part or async part. Standardize on `TaskCanceledException`?